server/lyft/decorators/audit_project_commands_wrapper.go

```diff
@@ -0,0 +1,140 @@
+package decorators
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/events"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"github.com/runatlantis/atlantis/server/lyft/aws/sns"
+)
+
+// AuditProjectCommandWrapper is a decorator that notifies sns topic
+// about the state of the command. It is used for auditing purposes
+type AuditProjectCommandWrapper struct {
+	SnsWriter sns.Writer
+	events.ProjectCommandRunner
+}
+
+func (p *AuditProjectCommandWrapper) Apply(ctx models.ProjectCommandContext) models.ProjectResult {
+	id := uuid.New()
+	startTime := time.Now()
+
+	applyEvent := &ApplyEvent{
+		Version:        1,
+		ID:             id.String(),
+		RootName:       ctx.ProjectName,
+		JobType:        models.ApplyCommand,
+		Respository:    ctx.BaseRepo.FullName,
+		Environment:    ctx.Tags["environment"],
+		PullNumber:     ctx.Pull.Num,
+		InitiatingUser: ctx.User.Username,
+		Project:        ctx.Tags["service_name"],
+		ForceApply:     ctx.ForceApply,
+		StartTime:      startTime.Format(time.RFC3339),
+		Revision:       ctx.Pull.HeadCommit,
+	}
+
+	if err := p.emit(ctx, ApplyEventInitiated, applyEvent); err != nil {
+		ctx.Log.Err("failed to emit apply event", err)
+	}
+
+	result := p.ProjectCommandRunner.Apply(ctx)
+
+	if result.Error != nil || result.Failure != "" {
+		if err := p.emit(ctx, ApplyEventError, applyEvent); err != nil {
+			ctx.Log.Err("failed to emit apply event", err)
+		}
+
+		return result
+	}
+
+	if err := p.emit(ctx, ApplyEventSuccess, applyEvent); err != nil {
+		ctx.Log.Err("failed to emit apply event", err)
+	}
+
+	ctx.Log.Info("sns topic has been updated with the apply state")
```

Comment:
Let's remove this. I want us be very strict about our logging going forward, since we have so many useless info logs in Atlantis that we need to delete.